### PR TITLE
#283: Successful identifier set in IdentifierCollection

### PR DESCRIPTION
**Pull request by Erwane**
get the successful identification provider from AuthenticationService or IdentifierCollection

### CODE REVIEW

File: src/Authenticator/Result.php [hunk 1]
```diff
@@ -113,4 +130,42 @@ public function getErrors()
     {
         return $this->_errors;
     }
+
+    /**
+     * Return null or the identifier who match the correct identity.
```
Comment (dereuromark) [inline src/Authenticator/Result.php]: The newlines are still missing in all docblocks.

**Comment by Erwane** [comment]
i've just saw this in AuthenticationService
```php
        foreach ($this->authenticators() as $authenticator) {
            $result = $authenticator->authenticate($request, $response);
            if ($result->isValid()) {
                $this->_successfulAuthenticator = $authenticator;
                $this->_result = $result;
```
I think my PR is useless ... 
i will take a break and check that.

**Comment by Erwane** [comment]
revamp my code and based it on the `AuthenticationService::getAuthenticationProvider`

Now, identifier is accessible thrue `AuthenticationService::getIdentificationProvider`, a convenient method for `IdentifierCollection::getIdentificationProvider`

**Comment by dereuromark** [comment]
Is anyone against this? Or can we merge?

**Comment by dereuromark** [comment]
We should also add some docs for this.

**Comment by markstory** [comment]
@dereuromark :+1: Done

**Comment by Erwane** [comment]
Thanks :)

**Pull request merged**